rfcs/proposed/numa_support/tbbbind-link-static-hwloc.org

```diff
@@ -0,0 +1,125 @@
+# -*- fill-column: 80; -*-
+
+#+title: Link ~tbbbind~ with static HWLOC to improve predictability of NUMA support API
+
+*Note:* This is a sub-RFC of the https://github.com/oneapi-src/oneTBB/pull/1535.
+Specifically, its section about "Increased availability of NUMA support".
+
+* Introduction
+oneTBB has a soft dependency on several variants of ~tbbbind~, which are loaded
+by the library as part of its initialization stage. In turn, each ~tbbbind~ has
+a hard dependency on a concrete version of the HWLOC library [1, 2]. The soft
+dependency of oneTBB on ~tbbbind~ allows the library to continue its execution
+even if the system loader is unable to resolve the hard dependency on HWLOC for
+~tbbbind~. In this case, the HW topology is not discovered and the machine is
+seen as if all CPU cores were uniform, which is the default TBB behavior when
+NUMA constraints are not used. Thus, the following code returns the values that
+do not reflect the real topology and do not matter:
+
+#+begin_src C++
+std::vector<oneapi::tbb::numa_node_id> numa_nodes = oneapi::tbb::info::numa_nodes();
+std::vector<oneapi::tbb::core_type_id> core_types = oneapi::tbb::info::core_types();
+#+end_src
+
+This lack of valid HW topology data due to absence of a third party library is
+the major problem with the current oneTBB behavior. There is no diagnostics for
+the issue, which likely makes it unnoticeable by developers, and the code that
+uses oneTBB NUMA support facilities continues running but does not use NUMA as
+intended.
+
+Having a dependency on a shared HWLOC library has advantages:
+1. Code reuse with all of the positive consequences out of this, including
+   relying on the same code that has been tested and debugged, allowing the OS
```

Comment:
but in fact most of Linux OSes has obsolete hwloc versions and relying on it does not provide benefits. 
IMO, having most up-to-date static HWLOC together with recent versions of oneTBB has benefits and fixes / new features are available to oneTBB immediately

Reply:
Thanks, I did not know that. I will consider this in the future changes.